Assets/Scripts/Screeps3D/RoomObjects/Views/ControllerView.cs

```diff
@@ -7,137 +7,122 @@ public class ControllerView : MonoBehaviour, IObjectViewComponent, IMapViewCompo
+        [SerializeField] private Renderer _l7 = default;
+        [SerializeField] private Renderer _l8 = default;
+        [SerializeField] private ScaleVisibility _progress = default;
         [SerializeField] private ScaleVisibility _vis = default;
         [SerializeField] private Collider _collider = default;
         [SerializeField] private ParticleSystem _ps = default;
-
-        private Texture2D _texture;
-        private Color _controllerWhite;
         private Controller _controller;
-        private Material _psMaterial;
-        private bool _isMyReservation;
+        private Ownership _ownership;
+        enum Ownership {
+            Me,
+            Enemy,
+            None
+        }
+        
+        private void setReservation() {
+            if(_controller?.ReservedBy?.Badge != null) {
+                _badge.materials[0].SetColor("EmissionColor", new Color(0.7f, 0.7f, 0.7f, 1f));
+                _badge.materials[0].SetTexture("EmissionTexture", _controller.Owner.Badge);
+                _badge.materials[0].SetFloat("EmissionStrength", 5);
+                _ownership = _controller.ReservedBy.UserId.Equals(Screeps_API.ScreepsAPI.Me.UserId) ? Ownership.Me : Ownership.Enemy;
+            }
+        }
 
-        private Color _myColor;
-        private Color _enemyColor;
-        public void Init()
-        {
-            InitTexture();
+        private void setOwnership() {
+            if (_controller?.Owner?.Badge != null) {
+                _badge.materials[0].SetColor("EmissionColor", new Color(0.7f, 0.7f, 0.7f, 1f));
+                _badge.materials[0].SetTexture("EmissionTexture", _controller.Owner.Badge);
+                _badge.materials[0].SetFloat("EmissionStrength", 5f);
+                _ownership = _controller.Owner.UserId.Equals(Screeps_API.ScreepsAPI.Me.UserId) ? Ownership.Me : Ownership.Enemy;
+            }
+        }
 
-            _psMaterial = _ps.GetComponent<Renderer>().material;//
-            _enemyColor = new Color(1.0f, 0.0f, 0.0f, 0.3f);
-            _myColor = new Color(0.0f, 1.0f, 0.0f, 0.3f);
-            changeReservationColor(false);
+        private void setParticleSystemColor() {
+            var isMy = false;
+            var psMain = _ps.main;
+            Color color = new Color(0.6f, 0.6f, 0.6f, 0.0f);
+            if(_ownership != Ownership.None) {
+                color = _ownership == Ownership.Me ? new Color(0.5f, 1.000f, 0.5f, 0.0f) : new Color(1.000f, 0.33f, 0.33f, 0.0f);
+            }            
+            psMain.startColor = color;
+            _core.materials[1].SetColor("EmissionColor", color);
+            _core.materials[1].SetFloat("EmissionStrength", 5f);
+        }
+        
+        private void customizeController() {
+            _ownership = Ownership.None;
+            _badge.materials[0].SetFloat("EmissionStrength", 0);
+            setReservation();
+            setOwnership();
+            setParticleSystemColor();
```

Comment:
clearing the ownership should probably only be done when there is no owner, e.g. a neutral controller. this method appears to be run every delta, on init and on load.